common/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java

```diff
@@ -480,12 +491,15 @@ private boolean sendReportWorker(boolean explicitRequest) {
     void addSpan(Span span) {
         lastNewSpanMillis.set(System.currentTimeMillis());
 
+        if (client == null) {
+            initializeCollectorClient();
```

Comment:
Right, the idea is only one of them will succeed in writing to the `client` variable, but I see your point re: this being wasteful.